identity-service/src/utils/configureAttester.js

```diff
@@ -0,0 +1,129 @@
+  if (!initialVals) {
+    initialVals = models.RewardAttesterValues.build()
+    initialVals.startingBlock = 0
+    initialVals.offset = 0
+    await initialVals.save()
+  }
+
+  const rewardsReporter = new RewardsReporter({
+    successSlackUrl: config.get('successAudioReporterSlackUrl'),
+    errorSlackUrl: config.get('errorAudioReporterSlackUrl'),
+    childLogger
+  })
+
+  // Init the RewardsAttester
+  const attester = new RewardsAttester({
+    libs,
+    logger: childLogger,
+    parallelization: config.get('rewardsParallelization'),
+    quorumSize: config.get('rewardsQuorumSize'),
+    aaoEndpoint,
+    aaoAddress,
+    startingBlock: initialVals.startingBlock,
+    offset: initialVals.offset,
+    challengeIdsDenyList,
+    reporter: rewardsReporter,
+    endpoints,
+    isSolanaChallenge: (challengeId) => challengeId === 'listen-streak',
+    runBehindSec,
+    updateValues: async ({ startingBlock, offset, successCount }) => {
+      childLogger.info(`Persisting offset: ${offset}, startingBlock: ${startingBlock}`)
+
+      await models.RewardAttesterValues.update({
+        startingBlock,
+        offset
+      }, { where: {} })
+
+      // If we succeeded in attesting for at least a single reward,
+      // store in Redis so we can healthcheck it.
+      if (successCount > 0) {
+        await redisClient.set(REDIS_ATTEST_HEALTH_KEY, Date.now())
+      }
+    },
+    getStartingBlockOverride: async () => {
+      // Retrieve a starting block override from redis (that is set externally, CLI, or otherwise)
+      // return that starting block so that the rewards attester changes its
+      // starting block, and then delete the value from redis as to stop re-reading it
+      const startBlock = await redisClient.get(REDIS_ATTEST_START_BLOCK_OVERRIDE_KEY)
+      if (startBlock === undefined || startBlock === null) {
+        return null
+      }
+
+      const parsedStartBlock = parseInt(startBlock, 10)
+      // Regardless if we were able to parse the start block override, clear it now
+      // so that subsequent runs don't pick it up again.
+      await redisClient.del(REDIS_ATTEST_START_BLOCK_OVERRIDE_KEY)
+
+      if (
+        parsedStartBlock !== undefined &&
+        parsedStartBlock !== null &&
+        !isNaN(parsedStartBlock)
+      ) {
+        return parsedStartBlock
+      }
+      // In the case of failing to parse from redis, just return null
+      return null
+    }
+  })
+
+  // Periodically check for new config and update the rewards attester
+  setInterval(async () => {
+    const { challengeIdsDenyList, endpoints, aaoEndpoint, aaoAddress, runBehindSec } = await getRemoteConfig(optimizely)
+    logger.info(`Pulled rewards attester remote config: endpoints ${endpoints}, aao ${aaoEndpoint} (${aaoAddress}), denyList: ${challengeIdsDenyList}, run behind: ${runBehindSec}`)
+    attester.updateConfig({ challengeIdsDenyList, endpoints, aaoEndpoint, aaoAddress, runBehindSec })
```

Comment:
The reason I originally thought to break it up into separate update methods is because i thought we might want to only modify one at a time and not recall the state of the other values to update, but i think this pattern is fine. no issues